OneDrive - w5fls/Desktop/potpie-main/potpie-main/app/modules/code_provider/local_repo/local_provider.py

```
@@ -0,0 +1,663 @@
+        if not ref:
+            ref = repo.active_branch.name
+
+        # Use git show to read file without checking out
+        # This avoids modifying working directory state
+        try:
+            file_content = repo.git.show(f"{ref}:{file_path}")
+        except GitCommandError as e:
+            if "does not exist" in str(e) or "path not in" in str(e):
+                raise FileNotFoundError(
+                    f"File not found: {file_path} at ref {ref}"
+                ) from e
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)